win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2994,39 +2988,102 @@ static unsigned int get_bezier_glyph_outline(FT_Outline *outline, unsigned int b
     return needed;
 }
 
-static INT
-IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG Width, LONG Height)
+static FT_Error
+IntRequestFontSize(PDC dc, PFONTGDI FontGDI, LONG lfWidth, LONG lfHeight)
 {
+    FT_Error error;
     FT_Size_RequestRec  req;
     FT_Face face = FontGDI->SharedFace->Face;
+    TT_OS2 *pOS2;
+    TT_HoriHeader *pHori;
+    FT_WinFNT_HeaderRec WinFNT;
+    LONG Ascent, Descent, Sum, EmHeight64;
 
-    if (Width < 0)
-        Width = -Width;
+    if (lfWidth < 0)
+    {
+        lfWidth = -lfWidth;
+    }
+    if (lfHeight == 0)
+    {
+        lfHeight = lfWidth;
+    }
+    if (lfHeight == 0)
+    {
+        lfHeight = 13;
+    }
+
+    pOS2 = (TT_OS2 *)FT_Get_Sfnt_Table(face, FT_SFNT_OS2);
+    pHori = (TT_HoriHeader *)FT_Get_Sfnt_Table(face, FT_SFNT_HHEA);
```

Comment:
I think for those calls freetype should be locked?
In that case, I think it would be nice if we assert (at the start of this function, or right where we use it) that the freetype lock is held:
`ASSERT_FREETYPE_LOCK_HELD()`

Reply:
OK.